src/Aspire.Dashboard/Components/Pages/Resources.razor.cs

```diff
@@ -441,18 +507,25 @@ private async Task OnToggleCollapse(ResourceGridViewModel viewModel)
     {
         // View model data is recreated if data updates.
         // Persist the collapsed state in a separate list.
-        if (viewModel.IsCollapsed)
-        {
-            viewModel.IsCollapsed = false;
-            _expandedResourceNames.Add(viewModel.Resource.Name);
-        }
-        else
+        viewModel.IsCollapsed = !viewModel.IsCollapsed;
+        _resourceNamesToExpanded[viewModel.Resource.Name] = !viewModel.IsCollapsed;
+
+        await LocalStorage.SetUnprotectedAsync(BrowserStorageKeys.ResourcesPageSettings, new ResourcesPageSettings(ResourceNamesToExpanded: _resourceNamesToExpanded));
```

Comment:
It's not terribly important, but we'd persist less data here if we just saved the list of resource names that had been collapsed.

Reply:
see above comment, changed to only do collapsed resources